js/ui/text_box/ui.text_editor.label.js

```diff
@@ -0,0 +1,61 @@
+import { getWidth, getOuterWidth } from '../../core/utils/size';
+import $ from '../../core/renderer';
+
+const TEXTEDITOR_WITH_LABEL_CLASS = 'dx-texteditor-with-label';
+const TEXTEDITOR_WITH_FLOATING_LABEL_CLASS = 'dx-texteditor-with-floating-label';
+const TEXTEDITOR_LABEL_CLASS = 'dx-texteditor-label';
+const TEXTEDITOR_LABEL_SELECTOR = '.' + TEXTEDITOR_LABEL_CLASS;
+const TEXTEDITOR_WITH_BEFORE_BUTTONS_CLASS = 'dx-texteditor-with-before-buttons';
+
+export function renderLabel({ editor, container, icon = null }) {
+    const labelElement = editor.$element().find(TEXTEDITOR_LABEL_SELECTOR);
+    const labelText = editor.option('label');
+    const labelMark = editor.option('labelMark');
+    const labelMode = editor.option('labelMode');
+
+    if(!editor.label && labelElement.length === 1 || labelElement.length === 2) {
+        labelElement.first().remove();
+    }
+
+    if(editor._$label) {
+        editor._$label.remove();
+        editor._$label = null;
+    }
+
+    editor.$element()
+        .removeClass(TEXTEDITOR_WITH_LABEL_CLASS)
+        .removeClass(TEXTEDITOR_WITH_FLOATING_LABEL_CLASS)
+        .removeClass(TEXTEDITOR_WITH_BEFORE_BUTTONS_CLASS);
+
+    if(!labelText || labelMode === 'hidden') return;
+
+    editor.$element().addClass(labelMode === 'floating' ? TEXTEDITOR_WITH_FLOATING_LABEL_CLASS : TEXTEDITOR_WITH_LABEL_CLASS);
+
+    const $label = editor._$label = $('<div>')
+        .addClass(TEXTEDITOR_LABEL_CLASS)
+        .html(`<div class="dx-label-before"></div><div class="dx-label"><span data-mark="${labelMark}">${labelText}</span></div><div class="dx-label-after"></div>`);
```

Comment:
I see several reasons to create a new `TextEditorLabel` class, not functions module:
1. Abstraction.
TextEditor calls some magic `renderLabel` public function and use `_$label` private variable after this. Which is this label life circle? What data is it based on? When `_$label` cache is cleared? How we will integrate resize observer?
Imho, label is a new ideological abstraction: it has own markup, props which describe component state, life circle.
2. Incapsulation.
- we mutate the `editor` parameter here :(
- `_$label` is created and configurated by this functions, but it's also used in ddEditor
3. Data passing.
There are 2 functions with the same parameter, which we pass one to another. It's a kind of a copypaste
4. Code style.
It's a consequence of the previous point. We have common data which is used in almost each operation => we cannot create one-purpose functions to improve readability because we don't wanna pass the same data to each new function => we have big functions with not strict responsibility.
5. Testing.
It's easier to test separate component than separate function (btw, what about tests?)